rewrite-java-11/src/main/java/org/openrewrite/java/Java11Parser.java

```diff
@@ -76,16 +70,21 @@
 
     private final Collection<NamedStyles> styles;
 
+    @Nullable
+    private final LoggingHandler loggingHandler;
+
     private Java11Parser(@Nullable Collection<Path> classpath,
                          Charset charset,
                          boolean relaxedClassTypeMatching,
                          boolean suppressMappingErrors,
                          boolean logCompilationWarningsAndErrors,
```

Comment:
Should this be an option set on `LoggingHandler`?

Reply:
I'm not sure. I intended LoggingHandler to be usable in potentially more cases than just Java11Parser, and this option is pretty specific to this situation. Also, I'm a bit confused by compilerLog. I don't see how it's used to do anything, it's just created, we set writers on it, and we reset it. Where do we log to it, or pass it into something that logs to it?

Reply:
We previously called logger.isWarnEnabled() at line 126 to avoid the overhead of allocating the log String. LoggingHandler being null is analogous to that in my mind. I know some dislike null checks, but this seems fairly clear to me. Let me know if you disagree.